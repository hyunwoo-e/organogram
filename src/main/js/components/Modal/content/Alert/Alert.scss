.Alert {
  width: 300px;
  padding: 25px;
  overflow: auto;
  text-align: center;
}

.Alert__title {
  text-align: center;
  font-weight: 600;
  color: $color-middle-gray-blue;
}

.Alert__content {
  font-weight: 400;
  font-size: 13px;
  color: $color-middle-gray-blue;
  display: block;
  text-align: center;
}

.Alert__btn-box {
  display: flex;
  justify-content: center;
}
