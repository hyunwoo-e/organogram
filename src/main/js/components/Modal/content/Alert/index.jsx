import React from 'react';
import PropTypes from 'prop-types';
import './Alert.scss';

const propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
  closeText: PropTypes.string,
  onModalClose: PropTypes.func.isRequired,
};
const defaultProps = {
  title: '',
  content: '',
  closeText: 'OK',
};

function Alert(props) {
  const {
    title,
    content,
    closeText,
    onModalClose,
  } = props;

  return (
    <div className="Alert">
      <h6 className="Alert__title space-2x"><strong>{title}</strong></h6>
      <span
        className="Alert__content space-4x"
      >
        {content.split('\n').map((line, index) => (
          <span key={index}>{line}<br /></span> // eslint-disable-line react/no-array-index-key
        ))}
      </span>
      <div className="Alert__btn-box">
        <button
          type="button"
          className="Btn Btn--primary Btn--sm"
          onClick={onModalClose}
        >
          {closeText}
        </button>
      </div>
    </div>
  );
}

Alert.propTypes = propTypes;
Alert.defaultProps = defaultProps;

export default Alert;
