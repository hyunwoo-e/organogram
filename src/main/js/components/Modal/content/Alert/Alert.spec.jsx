import React from 'react';
import { shallow } from 'enzyme';

import Alert from './';

describe('<Alert />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<Alert onModalClose={() => {}} />);

    expect(wrapper).toMatchSnapshot();
  });
});
