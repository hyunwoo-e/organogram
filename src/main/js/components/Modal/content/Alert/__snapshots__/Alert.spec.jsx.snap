// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should match to snapshot when render default 1`] = `
<div
  className="Alert"
>
  <h6
    className="Alert__title space-2x"
  >
    <strong />
  </h6>
  <span
    className="Alert__content space-4x"
  >
    <span
      key="0"
    >
      <br />
    </span>
  </span>
  <div
    className="Alert__btn-box"
  >
    <button
      className="Btn Btn--primary Btn--sm"
      onClick={[Function]}
      type="button"
    >
      OK
    </button>
  </div>
</div>
`;
